pkg/ui/workspaces/crdb-js-proto/defs.bzl

```diff
@@ -0,0 +1,112 @@
+load("@npm//pkg/ui/workspaces/crdb-js-proto:@bufbuild/buf/package_json.bzl", "bin")
+load("@rules_proto//proto:defs.bzl", "ProtoInfo")
+load("@aspect_rules_js//js:defs.bzl", "js_library")
+
+def buf_generate(name, out_dir_name, protos, **kwargs):
+    proto_target = "_%s_protos" % name
+    buf_target = "_%s_buf_bin" % name
+    _proto_sources(
+        name = proto_target,
+        protos = protos,
+    )
+    bin.buf(
+        name = buf_target,
+        srcs = [
+            ":node_modules",
+            proto_target,
+        ],
+        outs = [out_dir_name],
+        args = [
+            "generate",
+            "--template",
+            "{\"version\":\"v1\",\"plugins\":[{\"plugin\":\"es\",\"out\":\".\",\"path\":\"$(location :node_modules)/@bufbuild/protoc-gen-es/bin/protoc-gen-es\"}]}",
+            "--path",
+            # FAILS. proto_target expands to multiple paths and it is required to specify exactly one path per --path attribute
+            "$(execpaths %s)" % proto_target,
+            "--output $@",
+            "--debug --verbose",
+        ],
+        execution_requirements = {
+            "local": "1",
+        },
+        log_level = "debug",
+        chdir = "../../..",
+        copy_srcs_to_bin = False,
+    )
+    js_library(
+        name = name,
+        srcs = [
+            buf_target,
+        ],
+        **kwargs
+    )
+
+def protoc_gen(name, out_dir_name, protos, **kwargs):
+    proto_target = "_%s_protos" % name
+    genrule_target = "_%s_genrule" % name
+
+    _proto_sources(
+        name = proto_target,
+        protos = protos,
+    )
+
+    native.genrule(
+        name = name, # "_%s_genrule" % name,
+        outs = [out_dir_name],
+        cmd = "$(location @com_google_protobuf//:protoc) \
+              -Ibazel-out/darwin_arm64-dbg/bin/external/com_google_protobuf \
```

Comment:
Obviously, this cannot be checked in due to the hardcoding of `darwin_arm64-dbg`

Reply:
Can be resolved with $(BINDIR) as a prefix (ie `$(BINDIR)/external/com_google_protobuf`), but I hope it'll be possible to avoid these imports by using descriptor set files.

Reply:
@rickystewart , updated PR with working code. 
- It still requires `-I` imports to specify paths for .proto files. Unfortunately, it doesn't work when `$(GENDIR)/pkg` specified and needs to be more specific to avoid error when the same descriptors found in different paths.
- It's possible to generate this list of imports with following command on MACOS but doesn't work when specify in `genrule` (it doesn't return nested directories with `**`):
```
ls -d bazel-out/darwin_arm64-dbg/bin/pkg/**/_virtual_imports/*_proto
```
it is odd that this command doesn't rely on `SRCS` or `proto_target` label.